test/test_nn.py

```diff
@@ -8682,6 +8706,28 @@ def test_cudnn_noncontiguous_weight(self):
         self.assertEqual(F.conv1d(input, weights1, bias=None, stride=2, dilation=2),
                          F.conv1d(input, weights2, bias=None, stride=2, dilation=2))
 
+    @unittest.skipIf(not TEST_CUDA, "CUDA unavailable")
+    @unittest.skipIf(not TEST_CUDNN, "needs cudnn")
+    def test_conv_cudnn_nhwc(self):
+        input = torch.randint(1, 10, (2, 8, 4, 4), dtype=torch.float32, device="cuda").as_strided((2, 8, 4, 4), (128, 1, 32, 8), 0).requires_grad_(True)
+        grad = torch.rand(2, 4, 2, 2, dtype=torch.float32, device="cuda").as_strided((2, 4, 2, 2), (16, 1, 8, 4), 0)
+        conv = nn.Conv2d(8, 4, 3).cuda().float()
+
+        ref_input = input.detach().clone().contiguous().requires_grad_(True)
+        ref_grad = grad.detach().clone().contiguous()
+        ref_conv = nn.Conv2d(8, 4, 3).cuda().float()
+        ref_conv.load_state_dict(conv.state_dict())
+
+        out = conv(input)
+        out.backward(grad)
+        ref_out = ref_conv(ref_input)
+        ref_out.backward(ref_grad)
+
+        self.assertEqual(out, ref_out)
+        self.assertEqual(conv.weight.grad, ref_conv.weight.grad)
+        self.assertEqual(conv.bias.grad, ref_conv.bias.grad)
+        self.assertEqual(input.grad, ref_input.grad)
+
```

Comment:
Please also check:
if `out` is channels last `self.assertTrue(out.is_contiguous(memory_format=torch.channels_last))` 
if `ref_out` is contiguous (just to make sure we are promoting formats by mistake)
Same for gradients.

Reply:
Will do.